feat: add support for accessor arrays and refactor `stats/base/stdevwd`

Ari1009 opened the pull request:
Resolves #5685

## Description

> What is the purpose of this pull request?

This pull request:

-   add refactor and add protocol support to stats/base/stdevwd

## Related Issues

> Does this pull request have any related issues?

This pull request:

- resolve #5685

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Ari1009 committed:
accessor_ndarray

Ari1009 committed:
documentaion

Ari1009 committed:
Merge branch 'stdlib-js:develop' into accesor-sstdevwd

Ari1009 committed:
final_tes

Ari1009 committed:
final

aayush0325 reviewed (changes requested):
kindly address the CI failures and use #6248 as a reference PR, follow the conventions of benchmarks, examples docs, tests etc

aayush0325 commented on lib/node_modules/@stdlib/stats/base/stdevwd/docs/types/index.d.ts:
use tabs instead of spaces

aayush0325 commented on lib/node_modules/@stdlib/stats/base/stdevwd/docs/types/test.ts:
```diff
@@ -28,6 +28,17 @@ import stdevwd = require( './index' );
 
+// The function supports accessor arrays...
+{
+    const accessorArray = {
```
use `@stdlib/array/base/accessor` here and elsewhere

Ari1009 committed:
fix

Ari1009 commented:
Any updates

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into accesor-stdevwd

Ari1009 commented:
@aayush0325 can u give a recheck

gururaj1512 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into accesor-stdevwd

Ari1009 commented:
@aayush0325 can u give a recheck

gururaj1512 committed:
Merge branch 'develop' into accesor-stdevwd

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 committed:
fix: implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
chore: update benchmark script for improved clarity

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/stdevwd/docs/repl.txt:
```diff
@@ -109,8 +106,7 @@
 
     // Using offset parameter:
     > var x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ];
```
```suggestion
    > x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ];
```

kgryte committed:
docs: avoid redeclaration

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/stdevwd/test/test.main.js:
```diff
@@ -21,11 +21,11 @@
 var Float64Array = require( '@stdlib/array/float64' );
-var stdevwd = require( './../lib/stdevwd.js' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+var stdevwd = require( './../lib/main.js' );
```
```suggestion
var stdevwd = require( './../lib' );
```

kgryte committed:
test: update require path

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
